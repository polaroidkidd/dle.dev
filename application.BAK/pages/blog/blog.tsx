import React from "react";
import { blogEntriesURL } from "../../utils";
import Axios from "axios";
import ReactMarkdown from "react-markdown";
import { trackPromise } from "react-promise-tracker";
import { CodeBlock, LoadingIndicator } from "../../components";
import "./blog.scss";


const Blog: React.FC = () => {
  const [urls, serUrls] = React.useState<string[]>([]);
  const [articles, setArticles] = React.useState<string[]>([]);
  const [loading, setLoading] = React.useState<Boolean>(true);

  React.useEffect(() => {
    setTimeout(() => {
        trackPromise(
          Axios.get(blogEntriesURL).then((response) => {
            serUrls(response.data);
          })
        ).catch(e => console.error(e))
      }, 550
    )
  }, []);

  React.useEffect(() => {
    urls.forEach((url) => {
      trackPromise(Axios.get(url)
        .then((response) => {
          setArticles(((prevState) => [...prevState, response.data]));
          setLoading(false);
        })).catch(e => console.error(e));
    });
  }, [urls]);

  return (
    <>
      <div className="blog-container">
        {loading && <LoadingIndicator/>}
        {
          articles.map((article, index) => (
            <div className="markdown-body" key={index}>
              <ReactMarkdown
                source={article}
                renderers={{
                  code: CodeBlock,
                }}
              />
            </div>
          ))
        }

      </div>

    </>
  );
};
export { Blog };
