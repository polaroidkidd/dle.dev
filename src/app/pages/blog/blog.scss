@import "./src/app/styles/main";

.markdown-body {
  background-color: lighten($color-primary-background, 2%);
  border-radius: 0.5rem;
  margin: 2.5rem;
  padding: 2.5rem;
  width: 100%;

  & > h1 {
    margin: 2.75rem 0 1rem 0;
    @include heading-secondary
  }

  & > h2 {
    margin: 2.5rem 0 1rem 0;
    @include heading-tertiary
  }

  & > h3 {
    margin: 2.25rem 0 1rem 0;
    @include heading-tertiary
  }

  & > h4 {
    color: $color-primary-text;
    font-family: Poppins, sans-serif;
    font-size: 1.7rem;
    font-weight: 300;
    padding-left: 0.5rem;
  }

  & > h5 {
    color: $color-primary-text;
    font-family: Poppins, sans-serif;
    font-size: 1.7rem;
    font-weight: 300;
    padding-left: 0.5rem;

  }

  & > h6 {
    color: $color-primary-text;
    font-family: Poppins, sans-serif;
    font-size: 1.7rem;
    font-weight: 300;
    padding-left: 0.5rem;
  }

  & > p {
    color: $color-primary-text;
    font-family: Poppins, sans-serif;
    font-size: 1.7rem;
    font-weight: 300;
  }

  & > * strong {
    font-weight: 900;
  }

  & a {
    color: crimson;
    text-decoration: none;

    &:hover {
      color: lighten(crimson, 40%)
    }

    &:visited {
      color: lighten(crimson, 20%)
    }

    &:visited:hover {
      color: lighten(crimson, 40%)
    }
  }

  & ul {
    list-style: none;

    & li:before {
      color: crimson;
      content: "\203A";
      display: inline-flex;
      flex-direction: row;
      font-size: 2.5rem;
      justify-content: center;
      padding-left: 1.5rem;
      padding-right: 1rem;
    }

    & li {
      color: $color-primary-text;
      font-family: Poppins, sans-serif;
      font-size: 1.75rem;
      font-weight: 300;
    }
  }

  & ol {
    counter-reset: custom-couter;
    list-style: none;
    margin: 0;
    padding: 0;

    & li {
      counter-increment: custom-counter;
      @include paragraph;

      &::before {
        color: crimson;
        content: counter(custom-counter) ". ";
        font-family: monospace;
        font-size: 1.75rem;
        padding-left: 1.5rem;
        padding-right: 1rem;
      }
    }
  }

  & blockquote {
    border-left: 10px solid #CCC;
    margin: 1.5em 10px;
    padding: 0.5em 10px;
    quotes: "\201C" "\201D" "\201C" "\201D";

    &:before {
      color: #CCC;
      content: open-quote;
      font-size: 4em;
      line-height: 0.1em;
      margin-right: 0.25em;
      vertical-align: -0.4em;
    }


  }

  & blockquote p {
    display: inline;
  }

  & p {
    @include paragraph;
  }

}

