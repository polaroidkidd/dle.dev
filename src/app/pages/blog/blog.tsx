import React from "react";

const Blog: React.FC = () => {
  return (
    <div>BLOG CONTAINER</div>
  );
};
export { Blog};