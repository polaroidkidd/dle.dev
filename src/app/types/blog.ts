export type BlogEntryProperties = {
  published: string;
  title: string;
  url: string;
};
