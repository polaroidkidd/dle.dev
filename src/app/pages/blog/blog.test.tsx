import React from "react";
import renderer from "react-test-renderer";
import { Blog } from "./blog";

test("Render Blog Component", () => {
  const tree = renderer.create(
    <Blog/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
