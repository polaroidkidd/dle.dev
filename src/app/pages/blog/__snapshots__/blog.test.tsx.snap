// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Blog Component 1`] = `
<div>
  BLOG CONTAINER
</div>
`;
